opentelemetry-jaeger/src/lib.rs

```diff
@@ -382,11 +382,15 @@ fn links_to_references(links: &sdk::EvictedQueue<api::Link>) -> Option<Vec<jaege
 fn build_tags(span_data: &Arc<trace::SpanData>) -> Option<Vec<jaeger::Tag>> {
     let mut tags = Vec::with_capacity(span_data.attributes.len() + span_data.resource.len() + 4);
     let mut user_specified_error = false;
+    let mut user_specified_span_kind = false;
```

Comment:
Actually looking over this it may make sense to pull these bools out into a struct as we should probably do the same check for status code and status message.

Reply:
Does it make sense to store the keys in a `HashSet` as `str`s then before inserting checking whether the key is in the set?

Reply:
I tried something. Please let me know if it is similar to what you had in mind. Also `cargo fmt` ended up formatting some other files so I just pushed those changes in to. Thank you.

Reply:
Now that I see your explanation this makes so much more sense. Sorry about that. I think I was misunderstanding yesterday. I have made the requested change. Thank you.